tests/test-status.py

```diff
@@ -0,0 +1,23 @@
+import pytest
+import sh
+from utils import (
+    microk8s_reset,
+)
+
+
+class TestStatus(object):
+    @pytest.fixture(autouse=True)
+    def setup_cluster(self):
```

Comment:
I don't even know if that would hit the bug for which this PR has been opened, however you want to have an integration test for `microk8s.status` as well

Reply:
We already have an integration test for `microk8s status` [1].

You must have already seen that we only do integration tests and not unit tests, this is probably because we do not really have complex algorithms implemented.

Please do not continue with the refactoring you are proposing in this PR. Instead keep the patch to the `microk8s stauts` minimal (should be two lines) and extend the `test_help_text` to cover the bug you are hitting. In the test you would need to kill `kubelite` ad call `microk8s.status` before systemd restarts it.

[1] https://github.com/ubuntu/microk8s/blob/master/tests/test-addons.py#L57

Reply:
The suggestion of unit tests was outside the scope of this PR. However I didn't manage to create the integration test, also because it's very frustrating and insanely inefficient to develop without fast running unit tests. You should intensively rethink your stand towards unit tests and the need for it in this project. Maybe you add the integration test yourself - meanwhile I don't know if it makes a lot of sense.

Reply:
You are right, we should invest time in unit tests. Going forward we would like to rewrite some bash scripts in python. That would be a good time to pair our code with more tests.